### PR TITLE
add structured metadata to pages

### PR DESCRIPTION
**Pull request by zacjones93**
With next-seo it's fairly easy to add this structured data to each page. Using `SocialProfileJsonLd` we can add the instructor which links their twitter and website provided to that page.

On course pages we can add `CourseJsonLd`. This adds the provider (egghead) with title and description.

Article pages get `ArticleJsonLd` that includes author, publish date, publisher name, title, description, and images.

Lesson pages already had `VideoJsonLd` but added `SocialProfileJsonLd` as well.

### CODE REVIEW

File: src/components/layouts/collection-page-layout.tsx [hunk 1]
```diff
@@ -424,6 +424,18 @@ const CollectionPageLayout: React.FunctionComponent<
...
+        name={name}
+        url={`https://egghead.io/q/resources-by-${slug}`}
+        sameAs={[twitter, instructor.website]}
+      />
+      <CourseJsonLd
+        courseName={title}
+        providerName="egghead.io"
+        providerUrl="https://egghead.io"
+        description={truncate(removeMarkdown(description), {length: 155})}
+      />
```
Comment (zacjones93) [inline src/components/layouts/collection-page-layout.tsx:438]: course page

File: src/components/pages/lessons/lesson/index.tsx [hunk 2]
```diff
@@ -488,7 +488,7 @@ const Lesson: React.FC<React.PropsWithChildren<LessonProps>> = ({
   return (
     <>
       <NextSeo
-        description={removeMarkdown(description)}
+        description={truncate(removeMarkdown(description), {length: 155})}
```
Comment (zacjones93) [inline src/components/pages/lessons/lesson/index.tsx:491]: SEO doesn't like descriptions longer than 155 characters

File: src/components/pages/lessons/lesson/index.tsx [hunk 3]
```diff
@@ -511,10 +511,16 @@ const Lesson: React.FC<React.PropsWithChildren<LessonProps>> = ({
...
         uploadDate={lesson?.created_at}
         thumbnailUrls={compact([lesson?.thumb_url])}
       />
+      <SocialProfileJsonLd
+        type="Person"
+        name={instructor.full_name}
+        url={`https://egghead.io/${instructorPagePath}`}
+        sameAs={[`https://twitter.com/${instructor.twitter}`]}
+      />
```
Comment (zacjones93) [inline src/components/pages/lessons/lesson/index.tsx:523]: add instructor Person to lesson pages

File: src/components/search/instructors/instructor-essential.tsx [hunk 4]
```diff
@@ -50,6 +50,12 @@ const SearchInstructorEssential: FunctionComponent<
           ],
         }}
       />
+      <SocialProfileJsonLd
+        type="Person"
+        name={name}
+        url={`https://egghead.io/q/resources-by-${slug}`}
+        sameAs={[twitterHandle, instructor.website]}
+      />
```
Comment (zacjones93) [inline src/components/search/instructors/instructor-essential.tsx:58]: add instructor Person to profile pages

File: src/pages/blog/[slug].tsx [hunk 5]
```diff
@@ -83,6 +87,23 @@ const Tag = (props: any) => {
...
+        url={canonicalUrl}
+        title={title}
+        images={[ogImage]}
+        datePublished={publishedAt}
+        dateModified={updatedAt}
+        authorName={author.name}
+        description={truncate(removeMarkdown(seo.description), {length: 155})}
+        publisherName="egghead.io"
+        publisherLogo="https://res.cloudinary.com/dg3gyk0gu/image/upload/v1567198446/og-image-assets/eggo.svg"
+      />
```
Comment (zacjones93) [inline src/pages/blog/[slug].tsx:106]: Person and Article metadata for blog posts

**Review by joelhooks — APPROVED** [review]

**Pull request merged**